src/nbl/asset/IAssetManager.cpp

```diff
@@ -218,27 +218,50 @@ void IAssetManager::insertBuiltinAssets()
 			asset::ISpecializedShader::ESS_VERTEX,
-			{"nbl/builtin/material/lambertian/singletexture/specialized_shader.vert","nbl/builtin/material/debug/vertex_uv/specialized_shader.vert"});
+			{
+                "nbl/builtin/material/lambertian/singletexture/specialized_shader",
+                "nbl/builtin/material/debug/vertex_uv/specialized_shader",
+                "nbl/builtin/material/lambertian/singletexture/specialized_shader.vert",
+                "nbl/builtin/material/debug/vertex_uv/specialized_shader.vert"
+            });
 		buildInGLSLShader(fileSystem->loadBuiltinData<NBL_CORE_UNIQUE_STRING_LITERAL_TYPE("nbl/builtin/material/lambertian/singletexture/specialized_shader.frag")>(), // it somehow adds an extra "tt" raw string to the end of the returned value, beware
 			asset::ISpecializedShader::ESS_FRAGMENT, 
-			{"nbl/builtin/material/lambertian/singletexture/specialized_shader.frag"});
+			{
+                "nbl/builtin/material/lambertian/singletexture/specialized_shader",
+                "nbl/builtin/material/lambertian/singletexture/specialized_shader.frag"
+            });
 
-		// TODO
 		buildInGLSLShader(fileSystem->loadBuiltinData<NBL_CORE_UNIQUE_STRING_LITERAL_TYPE("nbl/builtin/material/debug/vertex_normal/specialized_shader.vert")>(),
 			asset::ISpecializedShader::ESS_VERTEX,
-			{"nbl/builtin/material/debug/vertex_normal/specialized_shader.vert"});
+			{
+                "nbl/builtin/material/debug/vertex_normal/specialized_shader",
+                "nbl/builtin/material/debug/vertex_normal/specialized_shader.vert"});
 		buildInGLSLShader(fileSystem->loadBuiltinData<NBL_CORE_UNIQUE_STRING_LITERAL_TYPE("nbl/builtin/material/debug/vertex_color/specialized_shader.vert")>(),
 			asset::ISpecializedShader::ESS_VERTEX,
-			{"nbl/builtin/material/debug/vertex_color/specialized_shader.vert"});
+			{
+                "nbl/builtin/material/debug/vertex_color/specialized_shader",
+                "nbl/builtin/material/debug/vertex_color/specialized_shader.vert"
+            });
 		buildInGLSLShader(fileSystem->loadBuiltinData<NBL_CORE_UNIQUE_STRING_LITERAL_TYPE("nbl/builtin/material/debug/vertex_uv/specialized_shader.frag")>(),
 			asset::ISpecializedShader::ESS_FRAGMENT,
-			{"nbl/builtin/material/debug/vertex_uv/specialized_shader.frag"});
+			{   
+                "nbl/builtin/material/debug/vertex_uv/specialized_shader",
+                "nbl/builtin/material/debug/vertex_uv/specialized_shader.frag"
+            });
 		buildInGLSLShader(fileSystem->loadBuiltinData<NBL_CORE_UNIQUE_STRING_LITERAL_TYPE("nbl/builtin/material/debug/vertex_normal/specialized_shader.frag")>(),
 			asset::ISpecializedShader::ESS_FRAGMENT,
-			{"nbl/builtin/material/debug/vertex_normal/specialized_shader.frag","nbl/builtin/material/debug/vertex_color/specialized_shader.frag"});
+			{
+                "nbl/builtin/material/debug/vertex_normal/specialized_shader",
+                "nbl/builtin/material/debug/vertex_color/specialized_shader",
```

Comment:
do not add cache key aliases without file extensions for specialized shaders

Reply:
sure

Reply:
corrected in following commit